main.tf

```diff
@@ -44,9 +44,16 @@ resource "aws_iam_role_policy" "iam_policy_for_lambda" {
 }
 POLICY
 }
+
+# Reference: https://github.com/hashicorp/terraform-provider-aws/issues/7385#issuecomment-756672006
```

Comment:
These changes were introduced in order to solve the problem where the pipeline would succeed, but the lambda function would not use the updated code.

The way this works is by leveraging the `source_code_hash` attribute of the `aws_lambda_function` resource.
The docs regarding this can be found [here](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/lambda_function#source_code_hash).

In particular, we are following an example found [here](https://github.com/hashicorp/terraform-provider-aws/issues/7385#issuecomment-756672006) that explains why and how we need to manage the source code hash while working with images from ECR.